### PR TITLE
Refactor `/mob/living/carbon/alien/humanoid` to be `/mob/living/carbon/alien/adult`

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This is a remake of #68911

- Changes `/mob/living/carbon/alien/humanoid` to be `/mob/living/carbon/alien/adult`
- Renames the `isalienhumanoid` define to be `isalienadult`
- Renames several files with alien humanoid in it to be alien adult.

## Why It's Good For The Game

Humanoid should be referring to the human species of mobs.

## Changelog

Not needed.

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
I don't really see the point to this

**Comment by ShizCalev** [comment]
Code churn.

**Comment by timothymtorres** [comment]
> image
> 
> I agree. This is just code churn.

Not really.

Code_aoSopVdaP2

This is just simple organization so that terms don't overlap when you are searching for "humanoid" when in 70% of cases it's referring to the human mob types.

**Comment by Mothblocks** [comment]
i think its a fine idea

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -91,19 +91,17 @@ GLOBAL_LIST_INIT(turfs_openspace, typecacheof(list(
 //More carbon mobs
 #define isalien(A) (istype(A, /mob/living/carbon/alien))
 
-#define isalienhumanoid(A) (istype(A, /mob/living/carbon/alien/humanoid))
-
 #define islarva(A) (istype(A, /mob/living/carbon/alien/larva))
 
-#define isalienadult(A) (istype(A, /mob/living/carbon/alien/humanoid) || istype(A, /mob/living/simple_animal/hostile/alien))
+#define isalienadult(A) (istype(A, /mob/living/carbon/alien/adult) || istype(A, /mob/living/simple_animal/hostile/alien))
```
Comment (Fikou) [inline code/__DEFINES/is_helpers.dm:96]: You cannot replace the first define with the second define, they are not equal and you will just be causing runtimes.

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (Mothblocks) [reply code/__DEFINES/is_helpers.dm:96]: why not?

Comment (timothymtorres) [reply code/__DEFINES/is_helpers.dm:96]: The old define I replaced was only used in 5 places:

Simple mob xenos are unable to generate this sound, no runtime.

https://github.com/tgstation/tgstation/blob/996b5a9c0df4e0a2fc4e8cb185e9c8863f660a0d/code/modules/mob/living/carbon/alien/emote.dm#L16-L30

Simple mob xenos are unable to swallow people, no runtime.

https://github.com/tgstation/tgstation/blob/45516f474147137d811d26ea72dd98a68953ad9f/code/modules/mob/living/carbon/alien/organs.dm#L316-L319

I tested spawning a bunch of slimes as a simple mob xeno, no runtime.

https://github.com/tgstation/tgstation/blob/70faa854b47fe784bb863e78a5b3a68f8464ab54/code/modules/mob/living/simple_animal/slime/life.dm#L345-L353

The queen early returns if trying to promote a simple mob that has no mind, so it doesn't runtime.

https://github.com/tgstation/tgstation/blob/bb4cce72e8d0d2121e8f8d084619435f2f50c920/code/modules/mob/living/carbon/alien/humanoid/queen.dm#L150-L159

Simple mob xenos have no runtimes for this either.

https://github.com/tgstation/tgstation/blob/43b00970496b9b4bce706b4a81503f769ffda8e5/code/game/machinery/doors/firedoor.dm#L122-L134

Comment (Mothblocks) [reply code/__DEFINES/is_helpers.dm:96]: i dont see how the two defines are not equivalent

**Comment by Mothblocks** [comment]
Please resolve conflicts, I think this is fine @timothymtorres

**Review by Mothblocks — CHANGES_REQUESTED** [review]
You are including several auto changelogs.

**Comment by timothymtorres** [comment]
This is ready to go.

**Pull request merged**